### ISSUE TITLE
[Prep] EC in EX: Build the foundations

### ISSUE THREAD

**Issue by manuroe** [vector-im/element-meta#1954]
### Description

Whatever UX decisions we need to be able to embed Element Call in a web view and connect it to the Widget API in the Rust SDK.
The Widget API implementation is a WIP but it is possible to work in parallel as suggested [here](https://github.com/vector-im/voip-internal/issues/145#issuecomment-1649952201).
For future purpose, the implementation of the webview should be considered for a generic widget, not only for call. We want more widgets in the future.

# Size estimate

M

# Dependencies

* https://github.com/vector-im/voip-internal/issues/193

# Acceptance criteria

* Element Call is opened in a fullscreen web view via an URL
* Element Call sends JSON events to the Widget API
* Element Call receives JSON events from the Widget API
* Permissions are asked as expected
* Element Call continues to work when the app is in bg or device locked
* Current audio playing is stopped when entering the call and resumed when leaving the call
* App can request Element Call to stop the call

# Out of scope

* Bundling the web app with the app package. We use a URL for now

# Subtasks

```[tasklist]
### Android
- [ ] https://github.com/vector-im/element-x-android/issues/1540
- [ ] https://github.com/vector-im/element-x-android/issues/1541
- [ ] https://github.com/vector-im/element-x-android/issues/1542
- [ ] https://github.com/vector-im/element-x-android/issues/1543
- [ ] https://github.com/vector-im/element-x-android/issues/1544
- [ ] https://github.com/vector-im/element-x-android/issues/1545
```

```[tasklist]
### iOS
 - [ ] https://github.com/vector-im/element-x-ios/issues/1442
 - [ ] https://github.com/vector-im/element-x-ios/issues/1443
 - [ ] https://github.com/vector-im/element-x-ios/issues/1444
 - [ ] https://github.com/vector-im/element-x-ios/issues/1445
 - [ ] https://github.com/vector-im/element-x-ios/issues/1446
 - [ ] https://github.com/vector-im/element-x-ios/issues/1447
```
